tests/unit/writers/test_writer_postgres.py

```
@@ -159,119 +178,84 @@ def test_postgres_test_write():
+def test_write_fails_when_connection_field_missing(reset_env, monkeypatch):
     writer = WriterPostgres({})
-    writer._db_config = {"database": "db", "host": "", "user": "u", "password": "p", "port": 5432}
-    ok, err = writer.write("public.cps.za.test", {})
-    assert not ok
-    assert "host" in err and "not configured" in err
+    monkeypatch.setattr(
+        type(writer),
+        "_pg_config",
+        property(lambda self: {"database": "db", "host": "", "user": "u", "password": "p", "port": 5432}),
+    )
+    with pytest.raises(WriteError, match="host.*not configured"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)